cuda/solver/lower_trs_kernels.cu

```diff
@@ -53,10 +58,124 @@ namespace cuda {
 namespace lower_trs {
 
 
+#if (defined(CUDA_VERSION) && (CUDA_VERSION > 9010))
+struct cusp_csrsm2_data {
+    int algorithm;
+    csrsm2Info_t solve_info;
+    cusparseSolvePolicy_t policy;
+    cusparseMatDescr_t factor_descr;
+    size_t factor_work_size;
+    void *factor_work_vec;
+};
+static cusp_csrsm2_data cusp_csrsm2_data{};
```

Comment:
Please don't use global variables, especially for data that is tied to an executor.
This prevents us from using multiple CUDA executors at the same time (while doing TRS).

Also: Why are you using the same name for the `struct` and the object?

I see 3 options:
- store the information inside the CUDA executor
- store it inside the LowerTrs class
- (also not a fan of it, but definitively better than the current state:) make a `std::map` with objects of the `*_data` structs tied to the corresponding executor.

Even the last option would be better than now, but I would prefer the first two options (or others that I did not think about).